ut.h

```diff
@@ -316,6 +316,11 @@ static inline char* double2str(double d, int* len)
 	return int2str_buf[buf];
 }
 
+#define BIGINT2STR_MAX_LEN  INT2STR_MAX_LEN
+static inline char* bigint2str(long long l, int* len)
+{
+	return int2bstr( l, int2str_buf[getstrbufindex()], len);
```

Comment:
It might be a problem here, as the `int2bstr()` function takes as input an unsigned long long - so the conversion will terribly go wrong if you have a negative big int value from DB.

Reply:
@bogdan-iancu, you are right!
I fixed this and tested with negative numbers.